pandera/hypotheses.py

```diff
@@ -310,6 +313,12 @@ def two_sample_ttest(
         4             4.0     B
 
         """
+        assert stats is not None, (
```

Comment:
`assert` statements should only be used for [debugging](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement) (and pytest). Assertions can be disabled on the python command line with the options [-O](https://docs.python.org/3/using/cmdline.html#cmdoption-o) and [-OO](https://docs.python.org/3/using/cmdline.html#cmdoption-oo), or by setting the environment variable [PYTHONOPTIMIZE](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONOPTIMIZE).

The usual practice is 
```python
try:
    from scipy import stats
except ImportError:
    _has_scipy= False
else:
    _has_scipy = True

...

if not _has_scipy:
    ...
```